core/device.cpp

```diff
@@ -252,3 +252,54 @@ extern "C" void create_fingerprint_node(struct fingerprint_table *table, uint32_
 		table->fingerprints.insert(it, fpr);
 	}
 }
+
+extern "C" void create_fingerprint_node_from_hex(struct fingerprint_table *table, uint32_t model, uint32_t serial,
+						const char *hex_data, uint32_t fdeviceid, uint32_t fdiveid)
+{
+	QByteArray raw = QByteArray::fromHex(hex_data);
+	create_fingerprint_node(table, model, serial,
+				(const unsigned char *)raw.constData(), raw.size(), fdeviceid, fdiveid);
+}
+
+extern "C" int nr_fingerprints(struct fingerprint_table *table)
+{
+	return table->fingerprints.size();
```

Comment:
Probably have to cast to int to avoid a conversion warning.
`(int)...` or `static_cast<int>(...)`

Reply:
I don't think I got a warning... but it would make more sense to make this an `unsigned int` function, don't you think?

Reply:
> I don't think I got a warning...

Yes, it seems one has to explicitly turn on `-Wconversion` to get a warning here.

> but it would make more sense to make this an unsigned int function, don't you think?

If you want to make it "really right", then this should return `size_t`, which is the standard type for lengths and indexes. On standard 64-bit x86, `unsigned int` is 4 bytes, whereas `size_t` is 8 bytes.

Reply:
Agreed. Yet... That's ridiculous overkill in this situation.